### PR TITLE
Enable i18n cache busting

### PR DESCRIPTION
**Pull request by thmsltmnn**
ng-jhipster now supports the "buildTimestamp" config parameter to enable cache busting when loading translation files. It is not used yet in generator-jhipster, this pull request should fix this.

During development, my solution works.

HOWEVER, when compiling the app with AOT, the string replacement does not work (or happens to late), so buildTimestamp always remains at the predefined value of '0' set in app.constants.ts.

If anyone knows how to fix this, help would be very much appreciated.

### CODE REVIEW

File: generators/client/templates/angular/src/main/webapp/app/shared/_shared-libs.module.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
 import { NgJhipsterModule } from 'ng-jhipster';
 import { InfiniteScrollModule } from 'ngx-infinite-scroll';
 import { CookieModule } from 'ngx-cookie';
+import {BUILD_TIMESTAMP} from '../app.constants';
```
Comment (deepu105) [inline generators/client/templates/angular/src/main/webapp/app/shared/_shared-libs.module.ts]: spacing inside braces

**Comment by sendilkumarn** [comment]
I will look into it later today or early tomorrow 👍

**Comment by sendilkumarn** [comment]
I think this is tricky, I confirm the issue. Can we try this with a simplified repo and check. I dont know whether the issue is with angular or ng-jhipster.

**Comment by sendilkumarn** [comment]
@thmsltmnn can we just use normal versioning with plain string here.

**Comment by deepu105** [comment]
It doesn't make any sense if the replacement for the other variables works how come this alone doesn't?

**Comment by thmsltmnn** [comment]
@sendilkumarn setting the version number manually as "buildTimestamp" works, and this is the workaround I use right now since it became possible to use this feature in ng-jhipster. This is not optimal, because it is another file you have to modify when increasing the version of your app, in addition to pom.xml and package.json.

@deepu105 I think the replacement works, but none of the other replaced variables goes into an NgModule configuration parameter, which seems to be treated differently by the Angular compiler.

**Comment by deepu105** [comment]
@thmsltmnn this issue should be solved by this https://github.com/jhipster/generator-jhipster/commit/a475422f52de85688b90bc7d63714852af1d8391
Can you try to add the buildTimestamp in the define section as I have done and try again

**Comment by deepu105** [comment]
@thmsltmnn I have updated your PR please test and confirm if it works with AOT

**Comment by thmsltmnn** [comment]
@deepu105 I tried it with a freshly generated JHipster 4.13.0 app: https://github.com/thmsltmnn/timely/commit/1eeee30477278846e668e0ed8d056589319cb48e

Sadly, it did not work. With AOT, the URL to request i18n files was http://localhost:8080/i18n/en.json?buildTimestamp=null

**Comment by deepu105** [comment]
@thmsltmnn i'm lost may be this is related to https://github.com/angular/angular/issues/18170

**Comment by deepu105** [comment]
@thmsltmnn btw hardcoding the value works but dynamic value doesn't so this is really an angular AOT issue similar to https://github.com/angular/angular/issues/18170
I hate angular AOT

This works
```
        NgJhipsterModule.forRoot({
            // set below to true to make alerts look like toast
            alertAsToast: false,
            i18nEnabled: true,
            buildTimestamp: '123',
            defaultI18nLang: 'en'
        }),
```

This doesn't
```
        NgJhipsterModule.forRoot({
            // set below to true to make alerts look like toast
            alertAsToast: false,
            i18nEnabled: true,
            buildTimestamp: new Date().getTime().toString(),
            defaultI18nLang: 'en'
        }),
```

**Comment by deepu105** [comment]
@thmsltmnn can you open an issue in Angular for this?

**Comment by deepu105** [comment]
@thmsltmnn a workaround would be to use `process.env.BUILD_TIMESTAMP` directly in the ng-jhipster lib instead of passing it via the module declaration, in theory it should work as its an env variable and webpack would replace it during compilation and should be AOT safe

**Comment by thmsltmnn** [comment]
@deepu105 Using a hardcoded value that I change on every release is the solution I use now. Might be good enough.

Another solution would be to move the translation stuff back from ng-jhipster to generator-jhipster, but I don't know if that fits together with the rest of the structure.

**Comment by sendilkumarn** [comment]
I don't think the second option is correct. Instead for the first, I would say yes. Till angular supports using symbol in here.

**Comment by deepu105** [comment]
I use the env variable in ng-jhipster directly, I dont think it needs to be configurable anyway as its only for cache busting so its good enough for this

**Pull request merged**